File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
       if not root: return 0  
       que = deque()
       if root: que.append(root)
       depth =0
       while len(que) > 0:
            depth+=1
            for i in range(len(que)):
                curr = que.popleft()
                if curr.right:que.append(curr.right)
                if curr.left: que.append(curr.left)
       return depth 

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int max_depth = 0;
    int maxDepth(TreeNode* root) {
        if (!root){return 0;}
        int left_max_depth = maxDepth(root->left);
        int right_max_depth = maxDepth(root->right);
        max_depth = max(max_depth,1 + left_max_depth+right_max_depth);
        
        return 1 + max(left_max_depth,right_max_depth);

    }
};